bin/prettier.js

```diff
@@ -122,14 +98,30 @@ function getIntOption(optionName) {
   process.exit(1);
 }
 
+function getTrailingComma() {
+  let trailingComma;
+  switch(argv["trailing-comma"]) {
+    case undefined:
+    case "none":
+      return "none";
+    case "":
+    case "es5":
+      return "es5";
+    case "all":
+      return "all";
+    default:
+      throw new Error("Invalid option for --trailing-comma");
+  }
```

Comment:
The old option works as much as we can make it work. The fundamental problem about turning this into a string option is that we can't have full backwards compatibility. This will never work:

```
prettier --trailing-comma foo.js
```

Because now the argument parser interprets `foo.js` as the option for `trailingComma`. The best we can do is error early with a clear message. There's just no way to get it to work as far as I know.

This will work:

```
prettier --trailing-comma --single-quote foo.js
```

Because it sees the option after it and does not include that as the argument, so the value of `trailing-comma` is `""` which I handle in the case statement.